src/tools/rustc-workspace-hack/Cargo.toml

```diff
@@ -72,25 +73,16 @@ features = [
 
 [dependencies]
 bstr = { version = "0.2.17", features = ["default"] }
-byteorder = { version = "1", features = ['default', 'std'] }
 clap = { version = "3.1.1", features = ["derive", "clap_derive"]}
 curl-sys = { version = "0.4.13", features = ["http2", "libnghttp2-sys"], optional = true }
-crossbeam-utils = { version = "0.8.0", features = ["nightly"] }
-libc = { version = "0.2.79", features = ["align"] }
 # Ensure default features of libz-sys, which are disabled in some scenarios.
 libz-sys = { version = "1.1.2" }
 # The only user of memchr's deprecated `use_std` feature is `combine`, so this can be
 # removed if/when https://github.com/Marwes/combine/pull/348 is merged and released.
 memchr = { version = "2.5", features = ["std", "use_std"] }
 # Ensure default features of regex, which are disabled in some scenarios.
 regex = { version = "1.5.6" }
-proc-macro2 = { version = "1", features = ["default"] }
-quote = { version = "1", features = ["default"] }
-rand_core_0_5 = { package = "rand_core", version = "0.5.1", features = ["getrandom", "alloc", "std"] }
-serde = { version = "1.0.82", features = ['derive'] }
 serde_json = { version = "1.0.31", features = ["raw_value", "unbounded_depth"] }
-smallvec = { version = "1.8.1", features = ['union', 'may_dangle'] }
-syn = { version = "1", features = ['fold', 'full', 'extra-traits', 'visit', 'visit-mut'] }
 url = { version = "2.0", features = ['serde'] }
```

Comment:
Probably not all removed crated should be removed, as some of them still used somewhere else.

Reply:
These were removed based on extensive testing to ensure they aren't used elsewhere.  There is some possibility that one of the more obscure platforms may enable one of these features, but that can be resolved via CI.

Reply:
Uhh, i don't understand. For example, `proc-macro2` with friends of `quote` and `syn` used literally everywhere: rustc\rustdoc\clippy\cargo\rustfmt.

Yes, removing this entries probably don't break things now, until some of crates decide to add\remove some of features (at which point CI will fail and contributor will be forced to return back here and rollback something).

Reply:
That is true of any shared dependency between the tools. Features can be adjusted at any time. The rustc-workspace-hack doesn't cover all of the shared dependencies (nor should it, since not all tools use the same dependencies).  I don't think rustc-workspace-hack should proactively add dependencies in the anticipation that they might be needed in the future. Adding new entries is relatively easy.  

As for proc-macro2 and quote, I doubt those will ever appear without their default features. It is not really clear from the history why those were added.